ros2sim/sims/arduino_mocker.py

```diff
@@ -0,0 +1,223 @@
+import pty
+import os
+import serial
+import threading
+import copy
+import logging
+
+from ros2sim.parsers import SimExecutor
+from ros2sim.parsers import SerialReadState
+from ros2sim.parsers import JointInformation
+
+
+class SerialSimulator:
+  def __init__(self, sim_executor: SimExecutor, baudrate=9600):
+    """Create a new Serial simulator
+
+    Args:
+      sim_executor (SimExecutor): Uses this to get information or make an action on 
+      solo_realtime environment
+    """
+    # Based on the data collected, appropriate actions will be taken using the parser
+    self.sim_executor = sim_executor
+    self.PREAMBLE_LENGTH = 4
+    self.DATA_BYTE_LENGTH = 2
+
+    self.num_joints = 8
+    self.imu_axes = 3
+
+    # Packet to store data temporarily as we read it. This data is not validated
+    self.temp_packet_data = JointInformation()
+    # After validation, data is copied here. For all intents and purposes, this
+    # is the only source of verified data
+    self.validated_packed_data = JointInformation()
+    # Initial state of the state machine
+    self.serial_read_state = SerialReadState.INIT
+    # List of all possible states for the state machine
+    self.list_of_states = [state for state in SerialReadState]
```

Comment:
is there no function in the enum that does this? i.e. `list(SerialReadState)`? genuinely have no idea.